components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/listener/ApplicationMgtAuditLogger.java

```diff
@@ -100,4 +109,186 @@ private String buildInitiatorUsername(String tenantDomain, String userName) {
         // Append tenant domain to username build the full qualified username of initiator.
         return UserCoreUtil.addTenantDomainToEntry(userName, tenantDomain);
     }
+
+    private String buildData(ServiceProvider serviceProvider) {
+
+        if (serviceProvider == null) {
+            return StringUtils.EMPTY;
+        }
+        StringBuilder data = new StringBuilder();
+        data.append("Name:").append(serviceProvider.getApplicationName()).append(", ");
+        data.append("Description:").append(serviceProvider.getDescription()).append(", ");
+        data.append("Resource ID:").append(serviceProvider.getApplicationResourceId()).append(", ");
+        data.append("Access URL:").append(serviceProvider.getAccessUrl()).append(", ");
+        data.append("Is Discoverable:").append(serviceProvider.isDiscoverable()).append(", ");
+        data.append("Is SaaS:").append(serviceProvider.isSaasApp()).append(", ");
+
+        if (serviceProvider.getInboundAuthenticationConfig() != null && ArrayUtils.isNotEmpty(serviceProvider
+                .getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs())) {
+            InboundAuthenticationRequestConfig[] requestConfigs = serviceProvider.getInboundAuthenticationConfig()
```

Comment:
Can this be null requestConfigs?

Reply:
this is already validated at line 126. 
`ArrayUtils.isNotEmpty(serviceProvider.getInboundAuthenticationConfig().getInboundAuthenticationRequestConfigs())`